src/pages/UpdateForm/UpdateForm.js

```diff
@@ -0,0 +1,525 @@
+    } else if (url.includes('rednoseday')) {
+      campaign = 'RND19';
+    } else {
+      campaign = 'CR';
+    }
+    return campaign;
+  }
+
+  /**
+   * Gets the current hostname.
+   * Replaces 'localhost' to a default or uses the browser's current url.
+   * @return string
+   */
+  getCurrentUrl() {
+    let url = null;
+    if (window.location.hostname === 'localhost') {
+      url = 'http://local.comicrelief.com';
+    } else {
+      url = window.location.href;
+    }
+    return url;
+  }
+
+  /**
+   * Updates validation state
+   * @param name
+   * @param valid
+   */
+  setValidity(childState, name) {
+    if (name && childState) {
+      this.setState((prevState) => {
+        let newState;
+        // If we already have saved validation object for this field
+        if (prevState.validation[name] !== undefined &&
+          // AND that object is empty OR the saved value isnt the same as our new childstate value
+          (prevState.validation[name].value === undefined ||
+            prevState.validation[name].value !== childState.value)) {
+          // .. then create a new state object, copying the current state, adding the
+          // current validation state plus the new value
+          newState = {
+            ...this.state,
+            validation: {
+              ...this.state.validation,
+              [name]: childState,
+            },
+          };
+        }
+        return newState;
+      });
+    }
+  }
+
+  setErrorMessagesToFalse() {
+    this.setState({
+      ...this.state,
+      showErrorMessages: false,
+    });
+  }
+
+  /**
+   * Goes through field refs, gets the first erroring field and focuses on it.
+   * If inputelement.labels is not supported: scrolls form into view
+   */
+  scrollToError() {
+    let item;
+    for (let i = 0; i <= this.fieldRefs.length; i += 1) {
+      item = this.fieldRefs[i];
+
+      // Customise this function for Radiobutton's markup
+      if (this.fieldRefs[i].nodeName === 'FIELDSET') {
+        // Gets the error div always added at the end
+        let lastChildErr = this.fieldRefs[i].children.length - 1;
+        lastChildErr = this.fieldRefs[i].children[lastChildErr].className.includes('error');
+        if (lastChildErr) {
+          item.scrollIntoView('smooth');
+          item.focus();
+          break;
+        }
+      } else if (this.fieldRefs[i].labels !== undefined) {
+        const classes = this.fieldRefs[i].labels[0].getAttribute('class');
+        if (classes.includes('error')) {
+          item.labels[0].scrollIntoView('smooth');
+          item.focus();
+          break;
+        }
+      } else {
+        document.querySelector('form').scrollIntoView();
+        break;
+      }
+    }
+    clearTimeout(scrollTimeout);
+  }
+
+  /**
+   * Maps the input field properties to a new array containing the input field instances
+   * @returns {Array}
+   */
+  createInputFields() {
+    const allFields = this.state.inputFieldProps;
```

Comment:
I think there's no need to use the state for this anymore as we can directly use the `defaultInputFieldsData` import right?

Reply:
Good point